### PR TITLE
Preserve environment_edf transform after edit.

### PR DESCRIPTION
**Pull request by pjessesco**
I added edit_entity() when Entity = EnvironmentEDF and ParentEntity = Scene.

### CODE REVIEW

File: src/appleseed.studio/mainwindow/project/projectbuilder.h [hunk 1]
```diff
@@ -165,15 +165,11 @@ Entity* ProjectBuilder::edit_entity(
     return new_entity_ptr;
 }
 
-template <>
-inline renderer::EnvironmentEDF* ProjectBuilder::edit_entity(
-    renderer::EnvironmentEDF*           old_entity,
-    renderer::Scene&                    parent,
-    const foundation::Dictionary&       values) const
+template <typename OldEntity, typename NewEntity>
+inline void copy_transform_sequence(
```
Comment (pjessesco) [inline src/appleseed.studio/mainwindow/project/projectbuilder.h]: Is it okay to define `copy_transform_sequence()` between `edit_entity()` definitions?

**Comment by pjessesco** [comment]
Similarly with this PR, same happens in `test scene/cameras/01` : blurry effect doesn't preserved with edit. I'd like to take it too, what value is related with blur effect?

File: src/appleseed.studio/mainwindow/project/projectbuilder.h [hunk 2]
```diff
@@ -164,15 +165,11 @@ Entity* ProjectBuilder::edit_entity(
     return new_entity_ptr;
 }
 
-template <>
-inline renderer::Camera* ProjectBuilder::edit_entity(
-    renderer::Camera*                   old_entity,
-    renderer::Scene&                    parent,
-    const foundation::Dictionary&       values) const
+template <typename OldEntity, typename NewEntity>
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/project/projectbuilder.h]: I would expect `OldEntity` and `NewEntity` to always be the same type; let's unify them in a single `Entity` param?

File: src/appleseed.studio/mainwindow/project/projectbuilder.h [hunk 3]
```diff
@@ -164,15 +165,11 @@ Entity* ProjectBuilder::edit_entity(
...
+inline void copy_transform_sequence(
+    const OldEntity&                   old_entity,
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/project/projectbuilder.h]: In the strict context of this function, I think `source_entity` and `destination_entity` would make more sense.

**Comment by dictoon** [comment]
> Similarly with this PR, same happens in test scene/cameras/01 : blurry effect doesn't preserved with edit. I'd like to take it too, what value is related with blur effect?

Good catch! That's a little surprising though. The blurring effect is due to depth of field which is entirely controlled by camera parameters (which are properly preserved across edits, normally):

    <parameter name="diaphragm_blades" value="5" />
    <parameter name="diaphragm_tilt_angle" value="0.0" />
    <parameter name="f_stop" value="1" />
    <parameter name="focal_distance" value="0.5" />

Investigation appreciated!!

**Pull request merged**